### PR TITLE
Use AWS accelerated AMIs

### PR DESCRIPTION
**Pull request by bd-g**
## Proposed changes

Previously, Amazon did not offer any official AMIs that included a recent version of the NVIDIA driver/toolkit and updated versions of system packages. The community demand for this has grown such that two weeks ago AWS released [an EKS accelerated AMI] based on Amazon Linux 2023. This means we now have an officially supported AWS AMI with bundled drivers and toolkits that is compatible with Deepgram workloads.

This PR updates the sample directory to use these new AMIs, and adjust the AWS example `values.yaml` file to no longer manually install NVIDIA drivers with the GPU operator Helm Chart. 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update or tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have tested my changes in my local self-hosted environment
  - I created a cluster using `eksctl v0.192.0` and deployed the Helm chart and verified the Engine container had access to the GPU.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: charts/deepgram-self-hosted/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
 
 ## [Unreleased]
 
+### Changed
+
+- AWS samples updated to take advantage of new [EKS accelerated AMIs](https://github.com/awslabs/amazon-eks-ami/releases/tag/v20240928), which bundle the required NVIDIA driver and toolkit instead of being installed by the NVIDIA GPU operator
```
Comment (jkroll-deepgram) [inline charts/deepgram-self-hosted/CHANGELOG.md:11]: I found this exact link not super enlightening - maybe link to the [top-level repo](https://github.com/awslabs/amazon-eks-ami), and/or their [changelog post](https://aws.amazon.com/about-aws/whats-new/2024/10/amazon-eks-nvidia-aws-neuron-instance-types-al2023/)?

File: charts/deepgram-self-hosted/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
 
 ## [Unreleased]
 
+### Changed
+
+- AWS samples updated to take advantage of new [EKS accelerated AMIs](https://github.com/awslabs/amazon-eks-ami/releases/tag/v20240928), which bundle the required NVIDIA driver and toolkit instead of being installed by the NVIDIA GPU operator 
+
```
Comment (jkroll-deepgram) [inline charts/deepgram-self-hosted/CHANGELOG.md:12]: Also, do any of these [setup instructions](https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html) apply such that we need to add more configuration steps to the Helm chart in some way?

File: charts/deepgram-self-hosted/CHANGELOG.md (hunk 1 above)
Comment (bd-g) [reply charts/deepgram-self-hosted/CHANGELOG.md:11]: That changelog post is *exactly* what I wanted, but my Google searches weren't finding that post so I fell back to the GH page.

Comment (bd-g) [reply charts/deepgram-self-hosted/CHANGELOG.md]: 3c0d68ad44b3

File: charts/deepgram-self-hosted/CHANGELOG.md (hunk 2 above)
Comment (bd-g) [reply charts/deepgram-self-hosted/CHANGELOG.md:12]: Those are manual instructions of pieces that are required, but are actually already handled by the `gpu-operator` Helm dependency we have.

More detailed description, the updated `values.yaml` file in this PR enables the `gpu-operator`, but specifically disables the driver and toolkit installation. Well, you may ask, what else is installed then if those two aren't installed? Answer is: the main thing is the NVIDIA device plugin. So our `gpu-operator` dependency handles those additional setup setps.

**Review by jkroll-deepgram — APPROVED** [review]

**Pull request merged**